vars/transportRequestUploadFile.groovy

```diff
@@ -79,12 +81,38 @@ def call(parameters = [:]) {
             }
         }
 
+
+        def transportRequestId = configuration.transportRequestId
+
+        if(transportRequestId?.trim()) {
+
+          echo "[INFO] Transport request id '${transportRequestId}' retrieved from parameters."
+
+        } else {
+
+          echo "[INFO] Retrieving transport request id from commit history [from: ${configuration.gitFrom}, to: ${configuration.gitTo}]." +
+               " Searching for pattern '${configuration.gitTransportRequestLabel}'. Searching with format '${configuration.gitFormat}'."
+
+            try {
+                transportRequestId = cm.getTransportRequestId(
+                                                  configuration.gitFrom,
+                                                  configuration.gitTo,
+                                                  configuration.gitTransportRequestLabel,
```

Comment:
For ```gitFrom```, ```gitTo```, ```gitTransportRequestLabel``` there are defaults in ```resources/default_pipeline_environment.yml```. If not that is a mistake and the default need to be inserted there.

Reply:
The step checks some properties (see code)
...
  .withMandatoryProperty('endpoint')
  .withMandatoryProperty('applicationId')
  .withMandatoryProperty('filePath')

does it mean that the gitXYZ arn't mandatory?

Reply:
There are mandatory, but for these parameters I know that there are reasonable defaults delivered with the lib (via resources/default_pipeline_enviroment.yml). Since there are defaults anyway, I don't check.

Side note: when switching to the nested properties (see #219) checks also for these properties are introduced since the filtering via ConfigurationHelper works only on top level entries the coding. I think we should make it explicit which parameters we use, hence I introduce ```withMandatoryProperty```. I still know there are defaults, but otherwise it wouldn't be that explicit.